01/app.js

```diff
@@ -1,15 +1,27 @@
 function getProductList() {
-    return [
-        {name: 'JavaScript: podstawy', type: 'book', count: 3, price: 67.19, discount: 0.21},
-        {name: 'React: podstawy', type: 'book', count: 4, price: 79.17, discount: 0.27}
+    return [{
+            name: 'JavaScript: podstawy',
+            type: 'book',
+            count: 3,
+            price: 67.19,
+            discount: 0.21
+        },
+        {
+            name: 'React: podstawy',
+            type: 'book',
+            count: 4,
+            price: 79.17,
+            discount: 0.27
+        }
     ]
 }
 
 function getTotalPrice(productList) {
-    return productList.reduce( (product, acc) => {
-        return acc + (product.price * product.count);
+    //Dzięki debuggerowi i dodaniu do watch zmiennej acc mogłem zauważyć, że wskazuje ona na cały obiekt, a nie jego poszczególne właściwości.
+    return productList.reduce((product, acc) => {
+        return ((acc.price * acc.count) * (1 - acc.discount) + (product.price * product.count) * (1 - product.discount)).toFixed(2);
```

Comment:
Sprawdź czy to rozwiązanie będzie działać jeśli tablica będzie pusta.
Oczywiście wtedy wynik działania tej funkcji to 0.